docs/devel/api_changes.md

```diff
@@ -0,0 +1,237 @@
+internal API at the center - every versioned API must be able to convert to the
+internal API, and vice-versa, but versioned APIs do not convert to other
+versioned APIs directly.  This sounds like a heavy process, but in reality we
+don't intend to keep more than a small number of versions alive at once.
+
+If you see errors from the `serialization_test`, it may indicate the need for
+explicit conversions.
+
+The conversion code resides with each versioned API -
+`pkg/api/<version>/conversion.go.  Unsurprisingly, this also requires you to
+add tests to `pkg/api/<version>/conversion_test.go`.
+
+## Update the fuzzer
+
+Part of our testing regimen for APIs is to "fuzz" (fill with random values) API
+objects and then convert them to and from the different API versions.  This is
+a great way of exposing places where you lost information or made bad
+assumptions.  If you have added any fields which need very careful formatting
+(the test does not run validation) or if you have made assumptions such as
+"this slice will always have at least 1 element", you may get an error or even
+a panic from the `serialization_test`.  If so, look at the diff it produces (or
+the backtrace in case of a panic) and figure out what you forgot.  Encode that
+into the fuzzer's custom fuzz functions.
+
+The fuzzer can be found in `pkg/api/testing/fuzzer.go`.
+
+## Update the semantic comparisons
+
+VERY VERY rarely is this needed, but when it hits, it hurts.  In some rare
+cases we end up with objects (e.g. resource quantites) that have morally
+equivalent values with different bitwise representations (e.g. value 0 with a
+base-2 formatter is the same as value 0 with a base-10 formatter).  The only
+way Go knows how to do deep-equality is through field-by-field bitwise
+comparisons.  This is a problem for us.
+
+The first thing you should do is try not to do that.  If you really can't avoid
+this, I'd like to introduce you to our semantic DeepEqual routine.  It lives in
+`pkg/api/helpers.go`.
+
+There's one other time when you might have to touch this: unexported fields.
+You see, while Go's `reflect` package is allowed to touch unexported fields, us
+mere mortals are not.  This includes semantic DeepEqual.  Fortunately, most of
+our API objects are "dumb structs" all the way down - all fields are exported
+(start with a capital letter) and there are no private fields.  But sometimes
+you want to include an object in our API that does have unexported fields
+somewhere in it (for example, time.Time has unexported fields).  If this hits
+you, you may have to touch the semantic DeepEqual customization functions.
+
+## Implement your change
+
+Now you have the API all changed - go implement whatever it is that you're
+doing!
+
+## Write end-to-end tests
+
+This is, sadly, still sort of painful.  Talk to us and we'll try to help you
+figure out the best way to make sure your cool feature keeps working forever.
+
+## Examples and docs
```

Comment:
We should mention description tag here. Every non-inline field must have a description tag. The tags are automatically pulled into the swagger model.

Presence of description tags is verified by hack/verify-description.sh, which also runs during pre-submit hooks and on Travis.